### PR TITLE
CR-4919 - Support using 2 repos for Kustomize apps

### LINKED ISSUE #80: Support using 2 repos for Kustomize apps (using the same token)
Currently, when creating a kustApp, the `apps/<app_name>/base` and `apps/<app_name>/overlays/<project_name>` directories are created in the GitOps repo (next to the `bootstrap` and `projects` directories).
With this new feature suggestion, an optional flag will be added to the `app create` command to supply a separate apps-repo. If the user adds this value, the apps folder structure will be created in this 2nd repo, while the GitOps repo will only contain the `config.json` file required by the project's ApplicationSet to create the Argo CD Application in the cluster.

The flag and feature is not relevant for dirApps.

### PR DESCRIPTION
**Pull request by ghost**
* added --apps-repo flag to app creation
* check if file exists on another repo
* bootstrap `repository.credentials` in config-map to use only host as prefix (usually `https://github.com`), to match any github repo that might require authentication.

### CODE REVIEW

File: docs/commands/argocd-autopilot_application_create.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ argocd-autopilot application create [APP_NAME] [flags]
 
 ```
       --app string                 The application specifier (e.g. github.com/argoproj/argo-workflows/manifests/cluster-install/?ref=v3.0.3)
+      --apps-git-token string      Your git provider api token [APPS_GIT_TOKEN]
+      --apps-repo string           Repository URL [APPS_GIT_REPO]
```
Comment (roi-codefresh) [inline docs/commands/argocd-autopilot_application_create.md:37]: it sucks that the descriptions must be the same...

File: pkg/application/application.go [hunk 2]
```diff
@@ -288,48 +288,50 @@ func newKustApp(o *CreateOptions, projectName, repoURL, targetRevision, repoRoot
...
 		// check if the bases are the same
 		log.G().Debug("application base with the same name exists, checking for collisions")
-		if collision, err := checkBaseCollision(repofs, baseKustomizationPath, app.base); err != nil {
+		if collision, err := checkBaseCollision(appsfs, baseKustomizationPath, app.base); err != nil {
 			return err
 		} else if collision {
 			return ErrAppCollisionWithExistingBase
 		}
+	} else if appsfs != repofs && repofs.ExistsOrDie(appPath) {
+		return ErrAppBaseOnDifferentRepo
```
Comment (roi-codefresh) [inline pkg/application/application.go:313]: don't know why I can't comment on the error definition itself but I think it should be more precise, not 
```
an application with the same name already exists in a different repo, consider choosing a different name
```
but 
```
an application with the same base exists on the bootstrap repo (or whatever we call the main repo)
```

Comment (roi-codefresh) [inline pkg/application/application.go:313]: don't know why I can't comment on the error definition itself but I think it should be more precise, not 
```
an application with the same name already exists in a different repo, consider choosing a different name
```
but 
```
an application with the same base exists on the bootstrap repo (or whatever we call the main repo)
```

File: pkg/application/application.go [hunk 3]
```diff
@@ -346,6 +348,10 @@ func kustCreateFiles(app *kustApp, repofs fs.FS, projectName string) error {
 	}
 
 	configPath := repofs.Join(overlayPath, "config.json")
```
Comment (roi-codefresh) [inline pkg/application/application.go:350]: doesn't the configPath always in the main repo?

**Review by roi-codefresh — APPROVED** [review]

File: pkg/application/application.go (hunk 3 above)
Comment (noam-codefresh) [reply pkg/application/application.go:350]: yes, that is why it is always being written to `repofs`

Comment (roi-codefresh) [reply pkg/application/application.go:350]: but this path is missing "overlays" no?

Comment (noam-codefresh) [reply pkg/application/application.go:350]: if the repofs == appsfs
  we put the config.json in `apps/<appName>/overlays/<projectName>/config.json`
else // repofs != appsfs
  we put the config.json in `apps/<appName>/<projectName>/config.json`

there is no need for an `overlays` folder if there is no matching `base` folder in the repofs (I think)

Comment (roi-codefresh) [reply pkg/application/application.go:350]: hmm, do we check if we try to add an application with the same name on the main repo (when the base was installed previously on some apps repo)?

File: pkg/application/application.go [hunk 4]
```diff
@@ -288,48 +288,50 @@ func newKustApp(o *CreateOptions, projectName, repoURL, targetRevision, repoRoot
...
-	appPath := repofs.Join(store.Default.AppsDir, app.Name())
-	basePath := repofs.Join(appPath, "base")
-	baseKustomizationPath := repofs.Join(basePath, "kustomization.yaml")
+	appPath := appsfs.Join(store.Default.AppsDir, app.Name())
+	basePath := appsfs.Join(appPath, "base")
+	baseKustomizationPath := appsfs.Join(basePath, "kustomization.yaml")
 
 	// check if base is in the same filesystem
-	if repofs.ExistsOrDie(baseKustomizationPath) {
+	if appsfs.ExistsOrDie(baseKustomizationPath) {
```
Comment (noam-codefresh) [inline pkg/application/application.go:304]: Need to check appPath

**Pull request merged**